### ISSUE TITLE
Maintainer teams for feedstocks

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-forge.github.io#40]
Not sure exactly how this works now so if I am off base, please feel free to correct my misunderstandings.

It seems that it would be very helpful to automatically add maintainers of feedstocks to conda forge GitHub teams (named after the feedstock) so they can easily merge PRs, deal with CI issues, and do other maintenance on the feedstock directly.

**Comment by pelson** [comment]
> It seems that it would be very helpful to automatically add maintainers of feedstocks to conda forge GitHub teams (named after the feedstock) so they can easily merge PRs, deal with CI issues, and do other maintenance on the feedstock directly.

Just to be clear, those listed under ``extra/recipe-maintainers`` *are* automatically added to the appropriate conda-forge teams (in https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/update_teams.py). It may be that you are asking for the *package* maintainers to be automatically added to the appropriate *feedstock* teams though (e.g. *all* matplotlib developers added to the conda-forge matplotlib feedstock maintainers). That is a somewhat harder problem which I haven't tried to address currently (as it requires some read permissions on teams that conda-forge does not have).

Looking at the logs from this repo, it turns out that the team script has been blocked by the failing feedstock submodule script (which we just fixed), and now the teams script also has an issue (looking into that now):

https://travis-ci.org/conda-forge/conda-forge.github.io/jobs/116310299

**Comment by jakirkham** [comment]
I was thinking `extra/recipe-maintainers` initially. Thanks for clarifying that this does in fact already happen. Sorry still learning how this all works. :)

Adding package maintainers is a nice idea, but I agree that is a much harder problem. Especially for things that are not on GitHub, have a repo name that doesn't nicely match their package name, etc. It seems the simple solution there is just to add people manually for now. Eventually, most things will be here and accurately match their maintainers so it will be a less relevant problem to solve.

**Comment by pelson** [comment]
> Sorry still learning how this all works. :)

No probs. This stuff isn't really documented at present. A gitter channel would have helped :+1:

**Comment by pelson** [comment]
The logs tell me this is now running as expected: https://travis-ci.org/conda-forge/conda-forge.github.io/jobs/116313298

You should have been invited to the tifffile maintainers team (looks like you've accepted too). How was the experience?

**Comment by jakirkham** [comment]
Yep, was just about to say it worked. Will close this.

**Comment by jakirkham** [comment]
Don't know why, but this seems to have gone awry with [qimage2ndarray]( https://github.com/conda-forge/qimage2ndarray-feedstock ).

**Comment by pelson** [comment]
I think it is just because https://travis-ci.org/conda-forge/conda-forge.github.io hasn't been triggered.
We currently don't have any infrastructure which can be cronned and rely on CI to update things. As a result, to update the teams, we need to trigger the CI by hand currently. I've done that now - qimage2ndarray should have a team when it gets through the queue.

**Comment by jakirkham** [comment]
So, I was looking for cron jobs and Travis CI and came across this issue ( https://github.com/travis-ci/travis-ci/issues/582 ), which led me to this ( https://nightli.es/ ). Maybe worth exploring.

**Comment by jakirkham** [comment]
Also, this might be interesting for scheduling more regular jobs ( http://tron-ci.herokuapp.com/ ).

**Comment by pelson** [comment]
> Also, this might be interesting for scheduling more regular jobs ( http://tron-ci.herokuapp.com/ ).

I had seen that (and had enabled on the conda-forge.github.io repo) but it didn't quite fit my mental model at the time. Might make sense to revisit that. Do you have experience of tron-ci?

**Comment by jakirkham** [comment]
I haven't used it before either. Just thinking it would be nice if this process sits somewhere that multiple core maintainers can check on.

**Comment by jakirkham** [comment]
I have spun this discussion off into a new issue ( https://github.com/conda-forge/conda-forge.github.io/issues/48 ).
